core-rust/state-manager/src/state_manager.rs

```diff
@@ -958,14 +938,7 @@ where
             committed_transaction_bundles.push((transaction, ledger_receipt, identifiers));
         }
 
-        let new_root_key = self.execution_cache.get(&parent_accumulator_hash).unwrap();
-        self.staged_store.reparent(new_root_key, &mut |key| {
-            self.execution_cache.remove_node(key);
-        });
-        self.execution_cache.set(
-            &parent_accumulator_hash,
-            StagedSubstateStoreKey::RootStoreKey,
-        );
+        self.execution_cache.progress_root(&parent_accumulator_hash);
```

Comment:
This should be next to the commit call imo.

In fact, now that this is a proper facade, this could even be part of the commit call?

Reply:
Although TBH I'd like to see:
* Store moved up under state manager
* Store managing its own internal RW Lock
* StagedStoreManager _not_ to have a store -- if absolutely necessary, it could have an Arc to it, although it probably doesn't need that

Reply:
The latter doesn't need to be part of this PR or workstream, but would be interested to hear your thoughts on it! 👍

Reply:
> This should be next to the commit call imo.

I am not sure what "commit call" means here.
But in general, I left this logic precisely in the same place where it was before (I strive to not change behavior during refactorings).
(just previously there were `.reparent()` + `.set()` calls and now they are bundled together under `.progress_root()`)

Regarding "where does root store belong" - I assumed that current state is the result of compromise with Rust The Language... I will have a look whether moving store's ownership is feasible (or makes sense conceptually) and we can of course discuss options. But this may be a refactoring unrelated to a hash tree.

And "Store managing its own internal RW Lock" - yup, totally a different work stream, I'd rather not be concerned with that "RW lock" thing here.